grr/server/grr_response_server/flows/general/osquery.py

```diff
@@ -67,13 +101,107 @@ def _UpdateProgress(
     self.state.progress.partial_table = _GetTruncatedTable(responses)
     self.state.progress.total_row_count = _GetTotalRowCount(responses)
 
+  def _GetPathSpecsToCollect(
+      self,
+      responses: flow_responses.Responses[rdf_osquery.OsqueryResult],
```

Comment:
Since you just iterate on responses, you can require `Iterable[rdf_osquery.OsqueryResult]` (and this is something `flow_responses.Responses` implements). It makes the code a bit simple and more reusable.